### PR TITLE
Html insert assets

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

**Comment by jbedard** [comment]
@googlebot I consent

File: e2e/ts_devserver/subpackage/BUILD.bazel [hunk 1]
```diff
@@ -12,14 +12,25 @@
...
+        "--html",
+        "subpackage/index.tmpl.html",
+        "--out",
+        "$@",
+        "--assets",
+        "$(location //:red-body-style.css)",
+    ],
+    data = [
+        "index.tmpl.html",
+        "//:red-body-style.css",
```
Comment (gregmagolan) [inline e2e/ts_devserver/subpackage/BUILD.bazel]: location not needed for `--html $(location :index.tmpl.html)`?

File: examples/kotlin/BUILD.bazel [hunk 2]
```diff
@@ -23,7 +22,7 @@ kt_js_library(
     deps = [":kotlinx-html-js"],
 )
 
-# Copy boostrap.js to the bin folder as _bootstrap.js
```
Comment (gregmagolan) [inline examples/kotlin/BUILD.bazel]: 👻

File: internal/node/npm_package_bin.bzl [hunk 3]
```diff
@@ -26,8 +26,13 @@ def _expand_location(ctx, s):
         # We'll write into a newly created directory named after the rule
         outdir_segments.append(ctx.attr.name)
 
+    if not ctx.attr.output_dir:
+        if s.find("$@") != -1 and len(ctx.outputs.outs) > 1:
+            fail("$@ substitution may only be used with a single out")
```
Comment (gregmagolan) [inline internal/node/npm_package_bin.bzl]: suggest to use `$(RULEDIR)` instead to make migration easier?

Comment (gregmagolan) [reply internal/node/npm_package_bin.bzl]: Oh nevermind. Now I noticed the `if not ctx.attr.output_dir:`

File: internal/node/npm_package_bin.bzl [hunk 4]
```diff
@@ -26,8 +26,13 @@ def _expand_location(ctx, s):
         # We'll write into a newly created directory named after the rule
```
Comment (gregmagolan) [inline internal/node/npm_package_bin.bzl]: Here would be a good place to help out users:
```
        if s.find("$@") != -1:
            fail("$@ substitution may not be used with output_dir set to True, use $(RULEDIR) instead")
```

File: e2e/ts_devserver/subpackage/BUILD.bazel (hunk 1 above)
Comment (alexeagle) [reply e2e/ts_devserver/subpackage/BUILD.bazel]: for inputs, `$(location foo)` is always the same as `package_name() + "/foo"` so I've been using interchangeably. I guess more consistency is fine

File: internal/node/npm_package_bin.bzl (hunk 3 above)
Comment (alexeagle) [reply internal/node/npm_package_bin.bzl]: sure but there's a common case where you had only one out and won't get this message even though you need to change. May as well add it though, could help someone

**Pull request merged**